tensorflow_addons/layers/crf.py

```
@@ -0,0 +1,334 @@
+                                            initializer=self.chain_initializer,
+                                            regularizer=self.chain_regularizer,
+                                            constraint=self.chain_constraint)
+
+        # bias that works with self.kernel
+        if self.use_bias:
+            self.bias = self.add_weight(shape=(self.units,),
+                                        name='bias',
+                                        initializer=self.bias_initializer,
+                                        regularizer=self.bias_regularizer,
+                                        constraint=self.bias_constraint)
+        else:
+            self.bias = 0
+
+        # weight of <START> to tag probability and tag to <END> probability
+        if self.use_boundary:
+            self.left_boundary = self.add_weight(shape=(self.units,),
+                                                 name='left_boundary',
+                                                 initializer=self.boundary_initializer,
+                                                 regularizer=self.boundary_regularizer,
+                                                 constraint=self.boundary_constraint)
+            self.right_boundary = self.add_weight(shape=(self.units,),
+                                                  name='right_boundary',
+                                                  initializer=self.boundary_initializer,
+                                                  regularizer=self.boundary_regularizer,
+                                                  constraint=self.boundary_constraint)
+
+        # or directly call self.built = True
+        super(CRF, self).build(input_shape)
+
+    def call(self, inputs, mask=None, **kwargs):
+        # mask: Tensor(shape=(?, ?), dtype=bool) or None
+
+        if mask is not None:
+            assert K.ndim(mask) == 2, 'Input mask to CRF must have dim 2 if not None'
```

Check using `if` and raise an Error.

---

What do you mean? Using if in here may raise an error?

---

I think he means you should raise an Error in else case, i.e; When `mask is None`

---

I think He means you should raise a ValueError (or another appropriate Error, like maybe AttributeError) when `mask is None`, i.e; 
```py
if mask is not None:
    assert K.ndim(mask) == 2, 'Input mask to CRF must have dim 2 if not None'
else:
    raise ValueError(A tensor for variable %s of type %s is required'
                         ' for layer %s' % (name, dtype.base_dtype, self.name))
```
name = mask
dtype.base_dtype = type(mask)
self.name = Arbitrary Name of the layer object (if any)

Something along theses lines, I suppose